### PR TITLE
Update xterm

### PR DESCRIPTION
**Pull request by hatched**
Xterm.js 3.x branch was updated to use canvas rendering and has a lot fewer style requirements making it considerably faster.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8659/

**Comment by hatched** [comment]
#### To QA

`guiproxy -env prod -flags terminal -config '{"jujushellURL": "wss://shell.jujugui.org/ws/"}'`

**Comment by huwshimi** [comment]
@hatched when I try and load the shell with those guiproxy settings I get: `Error talking to the terminal server: cannot log into juju: cannot authenticate user: interaction required but not possible` `Terminal connection unexpectedly closed.`

File: jujugui/static/gui/src/app/components/terminal/terminal.js [hunk 1]
```diff
@@ -41,10 +42,18 @@ class Terminal extends React.Component {
   */
   componentDidMount() {
     const props = this.props;
-    const term = new XTerm();
+    XTerm.applyAddon(fit);
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/terminal/terminal.js]: Nice, no more import side effects? Cool!

File: jujugui/static/gui/src/app/components/terminal/terminal.js [hunk 2]
```diff
@@ -41,10 +42,18 @@ class Terminal extends React.Component {
...
+    XTerm.applyAddon(terminado);
+    XTerm.applyAddon(WebfontLoader);
+    const term = new XTerm({
+      fontFamily: 'ubuntu mono',
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/terminal/terminal.js]: Nice!

**Comment by frankban** [comment]
QA is mostly ok except for one thing that we have to fix before landing this:
the focus is no longer on the term when you click to open it. This means that in production right now I can click on the shell icon, then wait, and when the term is ready I can just start typing. In this branch I have to further click to focus on the terminal.

**Comment by hatched** [comment]
Good catch @frankban that's been updated now.

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8661/

**Pull request merged**